azurerm/resource_arm_cosmos_db_account.go

```diff
@@ -398,7 +398,7 @@ func resourceAzureRMCosmosDBAccountFailoverPolicyHash(v interface{}) int {
 func validateAzureRmCosmosDBAccountName(v interface{}, k string) (ws []string, errors []error) {
 	value := v.(string)
 
-	r, _ := regexp.Compile("[a-z0-9-]")
+	r, _ := regexp.Compile("^([a-z0-9\\-]+)$")
```

Comment:
I can believe the escaping & quantifier (`+`) is necessary, but are the control characters like brackets & `^` & `$` too?
I mean, it applies to the whole string anyway and you don't need to capture anything via subgroup here?